src/Compilers/CSharp/Portable/CodeGen/EmitExpression.cs

```diff
@@ -1995,7 +1995,7 @@ private bool ConstructorNotSideEffecting(MethodSymbol constructor)
                 return true;
             }
 
-            if (originalDef.ContainingType.Name == TupleTypeSymbol.TupleTypeName &&
+            if (originalDef.ContainingType.Name == NamedTypeSymbol.TupleTypeName &&
```

Comment:
>TupleTypeName [](start = 67, length = 13)

Could be a (static) member of a static helper class `TupleTypeName`.

Reply:
I think the name is clear. Since we're leaving many other `TupleTypeSymbol` members in `NamedTypeSymbol`, I think it's okay to leave this one too. I prefer to leave as-is.

---
In reply to: [355072977](https://github.com/dotnet/roslyn/pull/39370#discussion_r355072977) [](ancestors = 355072977)